app/javascript/components/ReduxFormAutoComplete/index.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import { Dropdown, Menu, Item, Autocomplete, Field as GardenField } from '@zendeskgarden/react-dropdowns'
+import debounce from 'lodash.debounce'
+
+const ReduxFormAutocomplete = ({ dataSource, input: { onChange }, searchField }) => {
+  const searchData = dataSource.map(data => data[searchField])
+  // Selected Item is the String
+  const [selectedItem, setSelectedItem] = React.useState('')
+  const [inputValue, setInputValue] = React.useState('')
+  const [matchingOptions, setMatchingOptions] = React.useState(searchData)
+
+  /**
+   * Debounce filtering
+   */
+  const filterMatchingOptionsRef = React.useRef(
+    debounce(value => {
+      const matchingOptions = searchData.filter(data => {
+        return (
+          data
+            .trim()
+            .toLowerCase()
+            .indexOf(value.trim().toLowerCase()) !== -1
+        )
+      })
+
+      setMatchingOptions(matchingOptions)
+    }, 300)
+  )
+
+  React.useEffect(() => {
+    filterMatchingOptionsRef.current(inputValue)
+  }, [inputValue])
+
+  const renderOptions = () => {
+    if (matchingOptions.length === 0) {
+      return <Item disabled>No matches found</Item>
+    }
+
+    return matchingOptions.map(option => (
+      <Item key={option} value={option}>
+        <span>{option}</span>
+      </Item>
+    ))
+  }
+
+  return (
+    <Dropdown
+      inputValue={inputValue}
+      selectedItem={selectedItem}
+      onSelect={item => {
+        const itemData = dataSource.find(data => data[searchField] === item)
+        setSelectedItem(item)
+        onChange(itemData.id)
```

Comment:
what if `data[searchField] !== item`  `and itemData` is undefined? `onChange(itemData.id)` would throw a null ref exception I think. 
You can use Ramda to safely extract the id

```
onChange(R.prop('id', itemData))
```

Reply:
If it does happen, I suppose assigning it to `undefined` (no selection) is better than crashing the app, I like this, Ramda is so nice 😃

Reply:
I love ramda, the function composition features are awesome.